lib/web3auth_flutter.dart

```diff
@@ -0,0 +1,375 @@
+  final String? redirect_uri;
+
+  ExtraLoginOptions(
+      {this.additionalParams = const {},
+      this.domain,
+      this.client_id,
+      this.leeway,
+      this.verifierIdField,
+      this.isVerifierIdCaseSensitive,
+      this.display,
+      this.prompt,
+      this.max_age,
+      this.ui_locales,
+      this.id_token_hint,
+      this.login_hint,
+      this.acr_values,
+      this.scope,
+      this.audience,
+      this.connection,
+      this.state,
+      this.response_type,
+      this.nonce,
+      this.redirect_uri});
+}
+
+class Web3AuthOptions {
+  final String clientId;
+  final Network network;
+  final Uri? redirectUrl;
+  final String? sdkUrl;
+
+  Web3AuthOptions(
+      {required this.clientId,
+      required this.network,
+      this.redirectUrl,
+      this.sdkUrl});
+}
+
+class WhiteLabelData {
+  final String? name;
+  final String? logoLight;
+  final String? logoDark;
+  final String? defaultLanguage;
+  final bool? dark;
+  final HashMap? theme;
+
+  WhiteLabelData(
+      {this.name,
+      this.logoLight,
+      this.logoDark,
+      this.defaultLanguage,
+      this.dark,
+      this.theme});
+
+  Map<String, dynamic> toJson() {
+    return {
+      'name': name,
+      'logoLight': logoLight,
+      'logoDark': logoDark,
+      'defaultLanguage': defaultLanguage,
+      'dark': dark,
+      'theme': theme
+    };
+  }
+}
+
+class Web3AuthResponse {
+  final String privKey;
+  final TorusUserInfo userInfo;
+  final String? error;
+
+  Web3AuthResponse(this.privKey, this.userInfo, this.error);
+
+  @override
+  String toString() {
+    return "{privKey=$privKey, userInfo = ${userInfo.toString()}}";
+  }
+}
+
+class TorusUserInfo {
+  final String? email;
+  final String? name;
+  final String? profileImage;
+  final String? verifier;
+  final String? verifierId;
+  final String? typeOfLogin;
+  final String? aggregateVerifier;
+  final String? dappShare;
+
+  const TorusUserInfo({
+    required this.email,
+    required this.name,
+    this.profileImage,
+    this.verifier,
+    this.verifierId,
+    this.typeOfLogin,
+    this.aggregateVerifier,
+    this.dappShare,
+  });
+
+  @override
+  String toString() {
+    return "{email=$email, name=$name, profileImage=$profileImage, verifier=$verifier,"
+        "verifierId=$verifierId, typeOfLogin=$typeOfLogin}";
+  }
+}
+
+class UserCancelledException implements Exception {}
+
+class UnKnownException implements Exception {
+  final String? message;
+
+  UnKnownException(this.message);
+}
+
+class Web3AuthFlutter {
+  static const MethodChannel _channel = MethodChannel('web3auth_flutter');
+
+  static Future<String?> get platformVersion async {
+    final String? version = await _channel.invokeMethod('getPlatformVersion');
+    return version;
+  }
+
+  static Future<void> init(
+      {required String clientId,
+      Network? network,
+      required String redirectUri,
+      WhiteLabelData? whiteLabelData,
+      HashMap? loginConfig}) async {
```

Comment:
@tanujgupta what is supposed to be the type of value here? I suppose you should fill in the type paramters of this HashMap?

Reply:
@michaellee8 its <String, dynamic>. we are encoding the data into JSON to be parsed in the native layer. Look at the LoginConfig class.